recipes/server_check.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_file = '/etc/systemd/system/consul.service'
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0744'
+  action :create
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_rogue_server?(node.default['consul']['config']['retry_join'], node['ipaddress'])
+  # Stop the service if it is already running.
+  service 'consul' do
+    action :stop
```

Comment:
With this solution, simply rebooting the machine will not start the service. 

In order to bypass the systemd guardrail, need to remove the drop-in file at `/etc/systemd/system/consul.service.d/server_start_prevention.conf` .

In order to bypass the config file guardrail, need to overwrite `"server": true` in `/etc/consul/consul.json`

Then to allow server to listen on the desired DNS port, need to add 
SecureBits=keep-caps
AmbientCapabilities=CAP_NET_BIND_SERVICE
Capabilities=CAP_NET_BIND_SERVICE=+ep
CapabilityBoundingSet=CAP_SYSLOG CAP_NET_BIND_SERVICE

which the Chef cookbook would have done if `"server": true`.

Reply:
These extra guardrails ensure that if the rogue server start up again after accidental restarts.

Reply:
+1 to `disable` - I think the above makes sense, but I think a disabled systemd service is a cleaner expression that this service should not be running, instead of a failed attempt to start on reboot